partner_address_street3/tests/test_street_3.py

```diff
@@ -1,88 +1,45 @@
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2014 Nicolas Bessi, Alexandre Fayolle, Camptocamp SA
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
 import openerp.tests.common as test_common
+from ..hooks import uninstall_hook
 
 
 class TestStreet3(test_common.TransactionCase):
 
     def test_partner(self):
-        part_model = self.registry('res.partner')
-        country_model = self.registry('res.country')
-        country_id = country_model.create(
-            self.cr,
-            self.uid,
-            {
-                'name': 'Donut Land',
-                'code': 'DNL',
-            }
-        )
+        """"Test address_format has been updated on existing countries"""
+        us_country = self.env.ref('base.us')
 
-        self.assertTrue(country_id)
+        self.assertTrue('%(street3)s' in us_country.address_format)
 
-        create_data = {
+        homer = self.env['res.partner'].create({
             'name': 'Homer Simpson',
             'city': 'Springfield',
             'street': '742 Evergreen Terrace',
             'street2': 'Donut Lane',
             'street3': 'Tho',
-            'country_id': country_id,
-            'is_company': True
-        }
-
-        homer_id = part_model.create(
-            self.cr,
-            self.uid,
-            create_data
-        )
-
-        homer = part_model.browse(
-            self.cr,
-            self.uid,
-            homer_id,
-        )
-
-        self.assertEqual(
-            homer.country_id.address_format,
-            ("%(street)s\n%(street2)s\n%(street3)s\n"
-             "%(city)s %(state_code)s %(zip)s\n"
-             "%(country_name)s")
-        )
+            'country_id': us_country.id,
+        })
 
-        create_data = {
+        # test synchro of street3 on create
+        bart = self.env['res.partner'].create({
             'name': 'Bart Simpson',
-            'is_company': False,
             'parent_id': homer.id,
-            'use_parent_address': True
-        }
-
-        bart_id = part_model.create(
-            self.cr,
-            self.uid,
-            create_data
-        )
-
-        bart = part_model.browse(
-            self.cr,
-            self.uid,
-            bart_id,
-        )
-
-        self.assertTrue(bart.street3, 'Tho')
+            'type': 'contact',
+        })
+        self.assertEquals(bart.street3, 'Tho')
+
+        # test synchro of street3 on write
+        homer.write({'street3': 'in OCA we trust'})
+        self.assertEquals(bart.street3, 'in OCA we trust')
+
+    def test_hooks(self):
+        """"Test post_init_hook & uninstall_hook"""
```

Comment:
I agree with you in https://github.com/OCA/partner-contact/pull/316#discussion_r85279191

What I mean is that this test only tests the uninstall hook. Actually the post_init_hook is tested in every test because all of them would fail otherwise, but you don't specify it. This docstring should be "Test uninstall_hook", but if you simply name the method as `test_uninstall_hook`, then simply drop the docstring (it would be redundant) and it'll be OK :wink: